include/ry/runtime_alloc.hpp

```
@@ -0,0 +1,65 @@
+    if (!p) oom_abort(strlen(s) + 1);
+    return p;
+}
+
+inline char *checked_strndup(const char *s, size_t n) {
+    if (!s) return nullptr;
+    char *p = strndup(s, n);
+    if (!p) oom_abort(n + 1);
+    return p;
+}
+
+inline char *checked_memdup(const void *src, size_t len) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)